File: CS/Form1.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList;
using DevExpress.XtraReports.UI;


namespace E3239 {
    public partial class Form1 : Form {
        DataTable dt = new DataTable("TreeTable");
        TreeList treeList = new TreeList();

        public Form1() {
            InitializeComponent();

            LoadData();
            InitializeTreeList();
        }

        private void LoadData() {
            dt.Columns.Add("uniqueID", typeof(int));
            dt.Columns.Add("parentID", typeof(int));
            dt.Columns.Add("PositionNode", typeof(object));
            dt.Columns.Add("Code", typeof(string));
            dt.Columns.Add("Level", typeof(float));
            dt.Columns.Add("Position", typeof(float));

            for (int i = 0; i < 10; i++) {
                DataRow dr = dt.NewRow();
                dr["uniqueID"] = i;
                dr["parentID"] = i / 3;
                dr["Code"] = "code" + i;
                dr["Level"] = dr["Position"] = i;
                dt.Rows.Add(dr);
            }
        }

        private void InitializeTreeList() {
            Controls.Add(treeList);
            treeList.Dock = DockStyle.Fill;
            treeList.DataSource = dt;
            treeList.ParentFieldName = "parentID";
            treeList.KeyFieldName = "uniqueID";
            treeList.ExpandAll();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            Reports.TreeReport rep = new Reports.TreeReport(treeList.DataSource as DataTable);
            rep.ShowPreview();
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList


Namespace E3239
	Partial Public Class Form1
		Inherits Form
		Private dt As New DataTable("TreeTable")
		Private treeList As New TreeList()

		Public Sub New()
			InitializeComponent()

			LoadData()
			InitializeTreeList()
		End Sub

		Private Sub LoadData()
			dt.Columns.Add("uniqueID", GetType(Integer))
			dt.Columns.Add("parentID", GetType(Integer))
			dt.Columns.Add("PositionNode", GetType(Object))
			dt.Columns.Add("Code", GetType(String))
			dt.Columns.Add("Level", GetType(Single))
			dt.Columns.Add("Position", GetType(Single))

			For i As Integer = 0 To 9
				Dim dr As DataRow = dt.NewRow()
				dr("uniqueID") = i
				dr("parentID") = i \ 3
				dr("Code") = "code" & i
				dr("Position") = i
				dr("Level") = dr("Position")
				dt.Rows.Add(dr)
			Next i
		End Sub

		Private Sub InitializeTreeList()
			Controls.Add(treeList)
			treeList.Dock = DockStyle.Fill
			treeList.DataSource = dt
			treeList.ParentFieldName = "parentID"
			treeList.KeyFieldName = "uniqueID"
			treeList.ExpandAll()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			Dim rep As New Reports.TreeReport(TryCast(treeList.DataSource, DataTable))
			rep.ShowPreview()
		End Sub
	End Class
End Namespace